x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/workspace_panel.tsx

```diff
@@ -178,46 +181,70 @@ export function InnerWorkspacePanel({
   function renderVisualization() {
     useEffect(() => {
       // reset expression error if component attempts to run it again
-      if (expressionError) {
-        setExpressionError(undefined);
+      if (expression && localState.expressionBuildError) {
+        setLocalState(s => ({
+          ...s,
+          expressionBuildError: undefined,
+        }));
       }
     }, [expression]);
 
     if (expression === null) {
       return renderEmptyWorkspace();
     }
 
-    if (expressionError) {
+    if (localState.expressionBuildError) {
       return (
-        <EuiFlexGroup direction="column">
+        <EuiFlexGroup direction="column" className="lns">
           <EuiFlexItem data-test-subj="expression-failure">
-            {/* TODO word this differently because expressions should not be exposed at this level */}
+            <EuiIcon type="alert" size="xl" color="warning" />
             <FormattedMessage
               id="xpack.lens.editorFrame.expressionFailure"
-              defaultMessage="Expression could not be executed successfully"
+              defaultMessage="An error occurred in the expression"
             />
           </EuiFlexItem>
-          {expression && (
-            <EuiFlexItem>
-              <EuiCodeBlock>{toExpression(expression)}</EuiCodeBlock>
-            </EuiFlexItem>
-          )}
-          <EuiFlexItem>
-            <EuiCodeBlock>{JSON.stringify(expressionError, null, 2)}</EuiCodeBlock>
-          </EuiFlexItem>
+          <EuiFlexItem grow={false}>{localState.expressionBuildError}</EuiFlexItem>
         </EuiFlexGroup>
       );
-    } else {
-      return (
-        <ExpressionRendererComponent
-          className="lnsExpressionRenderer"
-          expression={expression!}
-          onRenderFailure={(e: unknown) => {
-            setExpressionError(e);
-          }}
-        />
-      );
     }
+
+    return (
+      <ExpressionRendererComponent
+        className="lnsExpressionRenderer"
+        expression={expression!}
+        renderError={(errorMessage?: string | null) => {
+          return (
+            <EuiFlexGroup direction="column" className="lns">
```

Comment:
This is an incomplete class, and to get it centered add `alignItems`
```suggestion
        <EuiFlexGroup direction="column" alignItems="center">
```